docs/current/guides/snippets/build-local-images/push/main.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"dagger.io/dagger"
+)
+
+func main() {
+	// initialize Dagger client
+	ctx := context.Background()
+	daggerClient, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stdout))
+	if err != nil {
+		panic(err)
+	}
+	defer daggerClient.Close()
+
+	// use NGINX container
+	// add new webserver index page
+	c := daggerClient.Container(dagger.ContainerOpts{Platform: "linux/amd64"}).
+		From("nginx:1.23-alpine").
+		WithNewFile("/usr/share/nginx/html/index.html", dagger.ContainerWithNewFileOpts{
+			Contents:    "Hello from Dagger!",
+			Permissions: 0o400,
+		})
+
+	// publish to local registry
+	addr, err := c.
+		Publish(ctx, "127.0.0.1:5000/my-nginx:1.0")
```

Comment:
Nit: With just one line to break, and while it doesn't exceed the max line width limit, I think it's better to keep these two in just one line.